backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/salesOrder.feature

```diff
@@ -84,4 +84,75 @@ Feature: sales order
+      | pp_67_5    | plv_67_2                          | p_32                    | 0.0      | PCE               | Normal                        |
+    And metasfresh contains PP_Product_BOMVersions:
+      | Identifier | M_Product_ID.Identifier | Name                |
+      | ppbv_67    | p_3                     | p_3_bomversion_name |
+    And metasfresh contains PP_Product_BOM:
+      | Identifier | M_Product_ID.Identifier | Name         | BOMType       | BOMUse        | C_UOM_ID.X12DE355 | PP_Product_BOMVersions_ID.Identifier | ValidFrom  |
+      | ppb_67     | p_3                     | p_3_bom_name | CurrentActive | Manufacturing | PCE               | ppbv_67                              | 2021-04-01 |
+    And metasfresh contains PP_Product_BOMLines:
+      | Identifier | PP_Product_BOM_ID.Identifier | M_Product_ID.Identifier | QtyBOM | C_UOM_ID.X12DE355 | ComponentType | ValidFrom  | Line |
+      | ppbl_67_1  | ppb_67                       | p_31                    | 3      | PCE               | CO            | 2021-04-01 | 100  |
+      | ppbl_67_2  | ppb_67                       | p_32                    | 4      | PCE               | CO            | 2021-04-01 | 200  |
+    And metasfresh contains C_BPartners:
+      | Identifier     | Name           | OPT.IsVendor | OPT.IsCustomer | M_PricingSystem_ID.Identifier |
+      | endcustomer_67 | Endcustomer_67 | N            | Y              | ps_3                          |
+      | vendor_67      | vendor_67      | Y            | Y              | ps_3                          |
+    And metasfresh contains C_BPartner_Products:
+      | C_BPartner_ID.Identifier | M_Product_ID.Identifier |
+      | vendor_67                | p_31                    |
+      | vendor_67                | p_32                    |
+    And metasfresh contains C_Orders:
+      | Identifier | IsSOTrx | C_BPartner_ID.Identifier | DateOrdered | POReference     | C_Payment_ID |
+      | o_3        | true    | endcustomer_67           | 2021-04-17  | po_ref_BOM_mock | 1000002      |
+    And metasfresh contains C_OrderLines:
+      | Identifier | C_Order_ID.Identifier | M_Product_ID.Identifier | QtyEntered |
+      | ol_3_1     | o_3                   | p_3                     | 10         |
+      | ol_3_2     | o_3                   | p_33                    | 10         |
+    And the order identified by o_3 is completed
+    And after not more than 10s, M_ShipmentSchedules are found:
+      | Identifier | C_OrderLine_ID.Identifier | IsToRecompute |
+      | s_ol_3     | ol_3_1                    | N             |
+    When generate PO from SO is invoked with parameters:
+      | C_BPartner_ID.Identifier | C_Order_ID.Identifier | PurchaseType | IsPurchaseBOMComponents |
+      | vendor_67                | o_3                   | Mediated     | true                    |
```

Comment:
mediated seems to be the wrong purchase order type. it's fine if mediated and IsPurchaseBOMComponents=true work together, but i can't imagine a use case, so better please test and make sure the "normal" bom-explode use-case works